aptos-move/mvhashmap/src/lib.rs

```diff
@@ -31,26 +34,37 @@ mod unit_tests;
 /// TODO: separate V into different generic types for data and modules / code (currently
 /// both WriteOp for executor, and use extract_raw_bytes. data for aggregators, and
 /// code for computing the module hash.
-pub struct MVHashMap<K, V: TransactionWrite, X: Executable> {
+pub struct MVHashMap<
+    K: ModulePath + Send + Sync + Hash + Clone + Eq + Debug,
+    V: TransactionWrite + Send + Sync,
+    X: Executable,
+> {
     data: VersionedData<K, V>,
     code: VersionedCode<K, V, X>,
 }
 
-impl<K: ModulePath + Hash + Clone + Eq + Debug, V: TransactionWrite, X: Executable>
-    MVHashMap<K, V, X>
+impl<
+        K: ModulePath + Hash + Clone + Eq + Send + Sync + Debug,
+        V: TransactionWrite + Send + Sync,
+        X: Executable,
+    > MVHashMap<K, V, X>
 {
     // -----------------------------------
     // Functions shared for data and code.
 
-    // Option<VersionedCode> is passed to allow re-using code cache between blocks.
-    pub fn new(code_cache: Option<VersionedCode<K, V, X>>) -> MVHashMap<K, V, X> {
+    // Executable cache is passed as it's re-used across blocks.
+    pub fn new(executable_cache: Arc<ExecutableStore<K, X>>) -> MVHashMap<K, V, X> {
         MVHashMap {
             data: VersionedData::new(),
-            code: code_cache.unwrap_or_default(),
+            code: VersionedCode::new(executable_cache),
```

Comment:
Let's move the cache outside. This way the flow for data and code will be asymmetric.

Reply:
yep, really good idea, doing this now.